aderyn_core/src/watchtower/lightchaser.rs

```diff
@@ -0,0 +1,614 @@
+    // Trigger Rate defined here - https://x.com/ChaseTheLight99/status/1750202276017283179?s=20
+    // NOTE - Here, the formula is slightly modified (we're not taking severity into account) because
+    // we would be grouping detectors by IssueSeverity so it doesn't become a factor
+    // Value to assign to severity is upto whoever is using this API
+    fn value(&self, detector_name: String) -> f64 {
+        let metrics = self.metrics_db.get_metrics(detector_name);
+        self.value_from_metrics(&metrics)
+    }
+
+    #[allow(clippy::let_and_return)]
+    fn value_from_metrics(&self, metrics: &Metrics) -> f64 {
+        let trigger_rate = metrics.trigger_count as f64 / metrics.experience as f64;
+        let lc_accuracy = metrics.lc_accuracy();
+        let detector_value = trigger_rate * lc_accuracy as f64; // min value = 0, max value = 1 * 5 = 5
+        let normalized_value = detector_value / IssueSeverity::COUNT as f64; // make it 0 to 1
+        normalized_value
+    }
+}
+
+impl GetsRegisteredDetectors for LightChaser {
+    fn get_registered_detectors_names(&self) -> Vec<String> {
+        self.metrics_db.get_all_detectors_names()
+    }
+}
+
+impl TagsTheDetector for LightChaser {
+    fn request_tag(&self, detector_name: String) -> Option<super::Tag> {
+        let metrics = self.metrics_db.get_metrics(detector_name.clone());
+        if metrics.is_acceptable() && !self.metrics_db.tags.contains_key(&detector_name) {
+            return None;
+        }
+        let mut tag_messages = vec![];
+        // Implicit tags (when accuracy drops too low for given severity)
+        if !metrics.is_acceptable() {
+            tag_messages.push(format!(
+                "{} is too low of an accuracy for {} detector",
+                metrics.lc_accuracy(),
+                metrics.current_severity
+            ));
+        }
+        // Explicit tags (set at the time of registering the detectors or ad hoc)
+        if self.metrics_db.tags.contains_key(&detector_name) {
+            if let Some(tags) = self.metrics_db.tags.get(&detector_name) {
+                tag_messages.extend(tags.messages.clone());
+            }
+        }
+        Some(super::Tag {
+            messages: tag_messages,
+        })
+    }
+
+    fn explicity_tag(&self, detector_name: String, message: String) {
+        self.metrics_db
+            .tag_detector_with_message(detector_name, message);
+    }
+
+    fn remove_tag(&self, detector_name: String) {
+        self.metrics_db.remove_tag(detector_name);
+    }
+}
+
+impl WatchTower for LightChaser {}
+
+#[cfg(test)]
+mod lightchaser_tests {
+    use serial_test::serial;
+    use strum::EnumCount;
+
+    use crate::{
+        detect::detector::{DetectorNamePool, IssueSeverity},
+        watchtower::{utils::MetricsDatabase, Feedback, InfersMetrics, WatchTower},
+    };
+
+    use super::LightChaser;
+
+    #[test]
+    #[serial]
+    fn lightchaser_can_register_new_detector() {
```

Comment:
I love seeing all these tests! 👏

Reply:
Tbh, I picked it up from Patrick's foundry course : ) I wrote so many tests trying to "follow along the video" in hopes to just finish fast and reach the next lesson...but in the meantime it was sooo many tests :P